icefoganalytics/internal-data-portal
# Add Minimal Docker/Docker Compose Node Setup

Fixes https://github.com/icefoganalytics/internal-data-portal/issues/3

# Context

Add just enough docker/docker compose + node so that the container boots, starts a node app, and returns a page so the deploy team can see the app is running.

# Implementation

Copy project setup from https://github.com/icefoganalytics/elcc-data-management

# Screenshots

404 page showing that the site booted correctly
![image](https://github.com/icefoganalytics/internal-data-portal/assets/23045206/cd4dba02-2a44-4c4e-8e6a-b57ff9d50732)

# Testing Instructions

1. Boot the app via `dev up`
2. Check that you can see the 404 page at http://localhost:3000/
3. Check that the http://localhost:3000/_status returns content appropriate for development
4. Stop the development app via `ctrl+c` then `dev down`
5. Test the production build via
    ```bash
    docker compose build --build-arg RELEASE_TAG=2024.01.8.1 --build-arg GIT_COMMIT_HASH=532bd759c301ddc3352a1cee41ceac8061bfa3f7
    ````
6. Start the prod build via `docker compose up`.
7. Check that you can see the 404 page at http://localhost:3000/
8. Check that the http://localhost:3000/_status page show content appropriate for production
